src/algebra/algebra/subalgebra.lean

```diff
@@ -1042,6 +1042,31 @@ lemma mem_center_iff {a : A} : a ∈ center R A ↔ ∀ (b : A), b*a = a*b := if
 
 end center
 
+section centralizer
+
+variables (R)
+
+/-- The centralizer of a set as a subalgebra. -/
+def centralizer (s : set A) : subalgebra R A :=
+{ algebra_map_mem' := λ r x mx, by rw [algebra.commutes],
```

Comment:
Can you extract this to `set.algebra_map_mem_centralizer`?

Reply:
Okay.